modules/Bio/EnsEMBL/Variation/TranscriptVariationAllele.pm

```diff
@@ -1698,6 +1698,10 @@ sub _get_fs_peptides {
   ### get CDS with alt variant
   my $alt_cds = $self->_get_alternate_cds();
   return undef unless defined($alt_cds);
+  unless ($alt_cds->seq =~ /([ACGT-])+/) {
```

Comment:
Should it be `$alt_cds->seq =~ /^([ACGT-])+$/)` to make sure there are only ATGC- characters in the CDS sequence ?

Reply:
I don't think so, with a sequence like NNNNANNNN or AAAANAAAA then we want it to carry on. This is just a crude 'if everything is Ns, stop' check.